File: old/LNode.java
public class LNode <E> extends Node <E> {
	protected LNode next;

	public LNode() {
		this.next = null;
		this.element = null;
	}

	public LNode(E element, LNode n) {
		this.element = element;
		this.next = n;
	}

	public LNode(E element) {
		this.element = element;
		this.next = null;
	}

	public LNode getNext() {
		return this.next;
	}
	
	public void  setNext(LNode n) {
		this.next = n;
	}
}

File: c++/oldy/older/LNode.cpp
#include <iostream>
#include "LNode.h"

using std::string;
using std::cout;
using std::endl;

int main() {
	LNode<string> n("David");
	cout << n << endl;

	LNode<string> n1 = LNode<string>();
	cout << n1 << endl;
	
	n1.setElement("Emily");
	cout << n1 << endl;
	
	n.setNext(&n1);

	LNode<string> &np = *n.next();	

	cout << n << endl;
	cout << np << endl;
	cout << *(n.next()) << endl;
	
	
}
